packages/main/src/Popup.ts

```diff
@@ -426,10 +426,14 @@ abstract class Popup extends UI5Element {
 			return;
 		}
 
-		const element = (this.getRootNode() as Document).getElementById(this.initialFocus)
-			|| document.getElementById(this.initialFocus)
-			|| await getFirstFocusableElement(this)
-			|| this._root; // in case of no focusable content focus the root
+		let element;
+
+		if (this.initialFocus) {
+			element = (this.getRootNode() as Document).getElementById(this.initialFocus)
+			|| document.getElementById(this.initialFocus);
+		} else {
+			element = await getFirstFocusableElement(this) || this._root; // in case of no focusable content focus the root
+		}
```

Comment:
If a wrong initial focus id is set, we still should focus the "first focusable element" or the root element. So probably this will be better:

```
let element;

		if (this.initialFocus) {
			element = (this.getRootNode() as Document).getElementById(this.initialFocus)
			|| document.getElementById(this.initialFocus);
		} 
		
element = element || await getFirstFocusableElement(this) || this._root; // in case of no focusable content focus the root
```